colour/plotting/common.py

```diff
@@ -29,25 +29,19 @@
 import contextlib
 import functools
 import itertools
+from dataclasses import dataclass, field
+from functools import partial
+
 import matplotlib.cm
 import matplotlib.pyplot as plt
 import matplotlib.ticker
 import numpy as np
-from cycler import cycler
-from dataclasses import dataclass, field
-from functools import partial
-from matplotlib.axes import Axes
-from matplotlib.colors import LinearSegmentedColormap
-from matplotlib.figure import Figure
-from matplotlib.patches import Patch
-from mpl_toolkits.mplot3d.axes3d import Axes3D
```

Comment:
Let's try to not reorder the imports automatically, it looks like `isort` work here and it is not doing a great job. We follow PEP8: https://peps.python.org/pep-0008/#imports

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.
> Imports should be grouped in the following order:
> 
> Standard library imports.
> Related third party imports.
> Local application/library specific imports.

Reply:
Yeah, it became clear to me that this was a way bigger tool / dev change than I was thinking originally. Removed in this PR. 

But I want to revisit this in the future. It's reading - pleasing when my auto-completed imports get put in the correct place by a tool. Ruff has some capability to do this (but I'm not sure about configurability) and isort can probably be made to match pep8.